extensions/cornerstone-dicom-sr/src/getSopClassHandlerModule.js

```diff
@@ -212,20 +212,26 @@ function _checkIfCanAddMeasurementsToDisplaySet(
     newDisplaySet
   );
 
-  for (let i = 0; i < images.length; i++) {
+  for (const imageId of imageIdsForDisplaySet) {
     if (!unloadedMeasurements.length) {
       // All measurements loaded.
-      break;
+      return;
```

Comment:
can't we move this `if (!unloadedMeasurements.length) {` outside?

Reply:
No. it can't.
`unloadedMeasurements` is being modified within the loop. (Line 241)